### PR TITLE
Switch to ruff as linter/formatter

### PR DESCRIPTION
**Pull request by badboy**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,11 +1,5 @@
 [bdist_wheel]
 python_tag = py3
 
-[flake8]
-exclude = docs
-max-line-length = 88
-# W503 - line break before binary operator - black formats it that way
-ignore = W503
```
Comment (Dexterp37) [inline setup.cfg:8]: Are these in a toml that was not uploaded?

Comment (badboy) [reply setup.cfg:8]: max-line length is default, ignore unneeded and the exclude unnecessary, because we explicitly pass the directories in our makefile

**Review by Dexterp37 — APPROVED** [review]
@badboy are we going to be ok with vendoring ruff in m-c?

**Comment by badboy** [comment]
> @badboy are we going to be ok with vendoring ruff in m-c?

First: [ruff is in use in m-c](https://searchfox.org/mozilla-central/rev/6b0b8c1003504e302e243c300355ef4d61e24909/tools/lint/python/ruff_requirements.in).
Second: we're not vendoring it. It's just a dev tool, specified in `requirements_dev.txt`, not as a direct dependency of the package.

**Pull request merged**